edweiss-app/app/(app)/courses/[id]/archive.tsx

```diff
@@ -48,37 +48,41 @@ export const testIDs = {
 const ArchiveScreen: ApplicationRoute = () => {
     const { rawAssignments } = useLocalSearchParams();
 
-    let assignments: AssignmentWithColor[] = [];
+    const assignments = useMemo<AssignmentWithColor[]>(() => {
+        if (typeof rawAssignments === 'string') {
+            try {
+                const parsedAssignments: AssignmentWithColor[] = JSON.parse(rawAssignments);
 
-    if (typeof rawAssignments === 'string') {
-        try {
-            assignments = JSON.parse(rawAssignments);
-            assignments.forEach((assignment: AssignmentWithColor) => {
-                assignment.dueDate = {
-                    seconds: assignment.dueDate.seconds,
-                    nanoseconds: assignment.dueDate.nanoseconds,
-                };
-            });
-        } catch (error) {
-            console.error('Failed to parse rawAssignments: ', error);
-            assignments = [];
+                // Mutate each assignment's dueDate property
+                parsedAssignments.forEach((assignment) => {
```

Comment:
I'm having a bit of trouble understand what this is supposed to do, you are reassigning the exact same values to.. themselves? Is there something I'm missing?
Also, you might want to look at `routeParameters` for automatic parsing and type safety